Source/WebCore/dom/Node.cpp

```diff
@@ -595,18 +598,42 @@ static RefPtr<Node> firstFollowingSiblingNotInNodeSet(Node& context, const HashS
 }
 
 // https://dom.spec.whatwg.org/#converting-nodes-into-a-node
-ExceptionOr<RefPtr<Node>> Node::convertNodesOrStringsIntoNode(FixedVector<NodeOrString>&& nodeOrStringVector)
+ExceptionOr<RefPtr<Node>> Node::convertNodesOrStringsOrTrustedScriptsIntoNode(RefPtr<Node> parent, FixedVector<NodeOrStringOrTrustedScript>&& vector)
 {
-    if (nodeOrStringVector.isEmpty())
+    if (vector.isEmpty())
         return nullptr;
 
     Ref document = this->document();
-    auto nodes = WTF::map(WTFMove(nodeOrStringVector), [&](auto&& variant) -> Ref<Node> {
-        return WTF::switchOn(WTFMove(variant),
-            [&](RefPtr<Node>&& node) { return node.releaseNonNull(); },
-            [&](String&& string) -> Ref<Node> { return Text::create(document, WTFMove(string)); }
-        );
-    });
+    NodeVector nodes;
+    for (auto& variant : vector) {
+        RefPtr<Text> text;
+        if (std::holds_alternative<String>(variant))
```

Comment:
Feels like there's probably a cleaner way to write this

Reply:
Is it possible to keep the WTF::switchOn structure and put the added functionality for TT in some helper function/method?

Reply:
I don't know that I can keep the WTF::switchOn structure, but I can definitely abstract some of this to a helper function in TrustedType.cpp. Will push up and see what you think.